### PR TITLE
Add a test for when content-length isn't available

### PR DESCRIPTION
**Pull request by jsumners**

### CODE REVIEW

File: formbody.js [hunk 1]
```diff
@@ -20,14 +20,27 @@ function formBodyPlugin (fastify, options, next) {
     if (contentLength > bodyLimit) return done(tooLargeError)
 
     let body = ''
-    req.on('error', done)
-    req.on('data', (data) => {
+    req.on('error', errorListener)
+    req.on('data', dataListener)
+    req.on('end', endListener)
+
+    function errorListener (err) {
```
Comment (allevo) [inline formbody.js]: Why not use `done` directly?

File: test/stream.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+const fork = require('child_process').fork
+const http = require('http')
+const path = require('path')
+const test = require('tap').test
+
+test('connection ends when limit reached', (t) => {
+  t.plan(5)
+  const proc = fork(path.join(__dirname, 'server.js'))
+
+  t.tearDown(() => proc.send('stop'))
```
Comment (allevo) [inline test/stream.test.js]: `proc.kill` maybe fits better.

Comment (jsumners) [reply test/stream.test.js]: Good call.

**Comment by nwoltman** [comment]
@jsumners This is tricky. I can make the `data` event happen after the response has been sent in my own application (using https) and only if I leave the `bodyLimit` at its default value. Maybe Node queues up reads on the request and then emits those as `data` events even after a response is sent.

I would keep the code that removes event listeners, but it's up to you. This seems very hard to test and may or may not affect a real production server.

**Comment by jsumners** [comment]
Can any other @fastify/fastify member shed some light? I really don't think it's necessary to remove the listeners because I'm fairly certain that as soon as `done(err)` is invoked then the request is terminated. So the stream should no longer be getting fed into the custom content parser. Thus, the accumulator only accumulates until `done(err)` is invoked. But I'm open to learning otherwise.

**Comment by nwoltman** [comment]
I think I figured it out. It has to do with sync vs async. If you call `done(err)` and the response is sent synchronously then the events stop. But if the response is sent asynchronously sometime later, it's possible for the events to continue. I think I was able to reproduce in my app because I have some async `onSend` hooks.

**Comment by jsumners** [comment]
Hmmm, that sounds like a core issue.

**Comment by nwoltman** [comment]
I added a commit that made tests pass for me locally. They're failing on CI though. @jsumners can you try seeing if the tests pass on your machine?

**Comment by jsumners** [comment]
@nwoltman nope. Still getting an ECONNRESET.

**Comment by nwoltman** [comment]
@jsumners After some digging I found that you were on the right track with [your comment](https://github.com/fastify/fastify/pull/645#issuecomment-356774831) in fastify/fastify#645. The reason calling `done(err)` multiple times doesn't do anything is because that makes `reply.send()` get called [here](), and since `reply.sent` will be `true` on every call after the first one, it will just log a warning that the response was already sent. I pushed a commit with a simpler test for this.

**Comment by jsumners** [comment]
Excellent. Looks good over here. Thank you for the assistance.

**Pull request merged**